client/post-editor/editor-post-formats/accordion.jsx

```diff
@@ -1,68 +1,68 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	pick = require( 'lodash/pick' ),
-	find = require( 'lodash/find' ),
-	classNames = require( 'classnames' ),
-	isEmpty = require( 'lodash/isEmpty' );
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { has, isEmpty } from 'lodash';
+import classNames from 'classnames';
 
 /**
  * Internal dependencies
  */
-var Accordion = require( 'components/accordion' ),
-	Gridicon = require( 'components/gridicon' ),
-	siteUtils = require( 'lib/site/utils' ),
-	PostFormats = require( './' );
+import Accordion from 'components/accordion';
+import Gridicon from 'components/gridicon';
+import QueryPostFormats from 'components/data/query-post-formats';
+import siteUtils from 'lib/site/utils';
+import PostFormats from './';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getPostFormats } from 'state/post-formats/selectors';
 
-module.exports = React.createClass( {
+const EditorPostFormatsAccordion = React.createClass( {
 	displayName: 'EditorPostFormatsAccordion',
 
 	propTypes: {
-		site: React.PropTypes.object,
-		post: React.PropTypes.object,
-		postFormats: React.PropTypes.arrayOf( React.PropTypes.shape( {
-			slug: React.PropTypes.string,
-			label: React.PropTypes.string
-		} ) )
+		site: PropTypes.object,
+		post: PropTypes.object,
+		postFormats: PropTypes.object
 	},
 
-	getFormatValue: function() {
-		if ( ! this.props.post ) {
+	getFormatValue() {
+		const { post, site } = this.props;
+		if ( ! post ) {
 			return;
 		}
 
-		if ( this.props.post.format ) {
-			return this.props.post.format;
+		if ( post.format ) {
+			return post.format;
 		}
 
-		return siteUtils.getDefaultPostFormat( this.props.site );
+		return siteUtils.getDefaultPostFormat( site );
 	},
 
-	getSubtitle: function() {
-		var postFormat;
+	getSubtitle() {
+		const formatValue = this.getFormatValue();
+		const { post, postFormats } = this.props;
 
-		if ( ! this.props.post || ! this.props.postFormats ) {
+		if ( ! post || ! postFormats ) {
 			return this.translate( 'Loading…' );
 		}
 
-		postFormat = find( this.props.postFormats, {
-			slug: this.getFormatValue() || 'standard'
-		} );
-
-		if ( postFormat ) {
-			return postFormat.label;
+		if ( has( postFormats, formatValue ) ) {
+			return postFormats[ formatValue ];
 		}
 
-		return this.translate( 'Standard', { context: 'Post format' } );
+		return this.translate( 'Standard', {
+			context: 'Post format'
+		} );
 	},
 
-	render: function() {
-		var classes = classNames( 'editor-post-formats__accordion', this.props.className, {
-			'is-loading': ! this.props.post || ! this.props.postFormats
+	render() {
+		const { className, post, postFormats } = this.props;
+		const classes = classNames( 'editor-post-formats__accordion', className, {
+			'is-loading': ! post || ! postFormats
 		} );
 
-		if ( isEmpty( this.props.postFormats ) ) {
+		if ( isEmpty( postFormats ) ) {
```

Comment:
Ah, that's a good point. And I think this PR is the right place for that - with a clear redux state the user will never reach the post formats query component, so the post formats might never load. Fixed and thanks for noticing 👍